dist/prototypes.py

```diff
@@ -53,42 +70,96 @@ def extract_prototypes(filename, regexp):
     return ret
 
 # Build function prototypes from a list of files.
-def fn_prototypes(fns, tests, name):
+def fn_prototypes(public_fns, private_fns, tests, name):
     for sig in extract_prototypes(name, r'\n[A-Za-z_].*\n__wt_[^{]*'):
-        fns.append(sig)
+        public_fns.append(sig)
 
     for sig in extract_prototypes(name, r'\n[A-Za-z_].*\n__wti_[^{]*'):
-        fns.append(sig)
+        private_fns.append(sig)
 
     for sig in extract_prototypes(name, r'\n[A-Za-z_].*\n__ut_[^{]*'):
         tests.append(sig)
 
-# Write results and compare to the current file.
-# Unit-testing functions are exposed separately in their own section to
-# allow them to be ifdef'd out.
-def output(fns, tests, f):
-    tmp_file = '__tmp_prototypes' + str(os.getpid())
+# Create a new header file with function declarations.
+# As this is a new file also generate boilerplate like #pragma once
+def create_new_header(tmp_file, fns, tests, private_file=""):
     tfile = open(tmp_file, 'w')
+    tfile.write(COPYRIGHT)
     tfile.write("#pragma once\n\n")
+
+    if private_file:
+        tfile.write(f"#include \"{private_file}\"\n\n")
+
+    tfile.write("/* DO NOT EDIT: automatically built by prototypes.py: BEGIN */\n\n")
+
     for e in sorted(list(set(fns))):
```

Comment:
what if fns is empty for _public.h?

Reply:
I thought about this during dev and came to the conclusion a module with no public functions is dead code 🤷.
I can add a comment if preferred

Reply:
it's okay, I will resolve this.